Feature/administration products and categories

whaberek opened the pull request.

Commit:
feature/connect_with_firebase
working dialog box for login

Commit:
feature/connect_with_firebase
Working initialize app firebase

Commit:
feature/connect_with_firebase
Working connect to firebase project with email and password

Commit:
feature/connect_with_firebase
Now we can log in and log out in this application usong firebase / add toast alert

Commit:
feature/connect_with_firebase
hide 'logout' button when user logged out and show 'sign in' button

Commit:
feature/connect_with_firebase
Loading bar / Menu for logged user only

Commit:
feature/connect_with_firebase
Working pages for admin and change location when user is not logged

Commit:
feature/connect_with_firebase
Change after code review

Commit:
feature/connect_with_firebase
Remove console log

Commit:
feature/connect_with_firebase
Put catch in a new line in login component

Commit:
feature/connect_with_firebase
Change folder structure and add view for product and categories list

Commit:
feature/connect_with_firebase
Add pagination for products in admin panel

Commit:
feature/connect_with_firebase
Features with search product by title and category

Commit:
feature/connect_with_firebase
Change data type

Commit:
feature/administration_products_and_categories
Create and delete categories

Commit:
feature/administration_products_and_categories
Get data from firbase for products

Commit:
feature/administration_products_and_categories
Working realtime database, delete products and category \ Change all http get data to firebase method

Commit:
feature/administration_products_and_categories
Working create-product page / Create first working get categoris function in services with response data

Commit:
feature/administration_products_and_categories
All logic in application working on firebase method in services

Commit:
feature/administration_products_and_categories
Remove not use pagination from project

mtrzesniowski commented on app/pages/admin/categories/create/category-create.component.js:
```diff
@@ -0,0 +1,33 @@
+    redirectUser() {
+        this.firebase.auth().onAuthStateChanged(user => {
+            if (!user) {
+                this.$location.path('/');
```
Maybe better call `$state.go` ?

mtrzesniowski commented on app/pages/admin/categories/list/categories.component.js:
```diff
@@ -0,0 +1,67 @@
+    redirectUser() {
+        this.firebase.auth().onAuthStateChanged(user => {
+            if (!user) {
+                this.$location.path('/');
```
Maybe better use `$state.go` ?

mtrzesniowski commented on app/pages/admin/products/create/product-create.component.js:
```diff
@@ -0,0 +1,42 @@
+    redirectUser() {
+        this.firebase.auth().onAuthStateChanged(user => {
+            if (!user) {
+                this.$location.path('/');
```
Maybe better use `$state.go` ?

Commit:
feature/administration_products_and_categories
Other method for redirect user(change from $location to $state.go) / Add filter for products list on admin panel

Commit:
feature/administration_products_and_categories
Create possibility for edit categories name

Commit:
feature/administration_products_and_categories
Repair bug with disable button in edit products / add license files

Commit:
feature/administration_products_and_categories
Remove double information about licence

Commit:
feature/administration_products_and_categories
The ability to change your display name and avatar

Commit:
feature/administration_products_and_categories
remove duplicate style from css / set overflow visible for grid view

Commit:
feature/administration_products_and_categories
improving the appearance for mobile devices

Commit:
feature/administration_products_and_categories
Environment prepared for the unit tests

mtrzesniowski commented on app/components/app/app.service.js:
```diff
@@ -1,29 +1,64 @@
+    }
+
+    getProductsByCategory(category) {
+        const dbRefObjectProducts = this.firebase.database().ref('api/v1').child('products').orderByChild('category').equalTo(category);
```
In case of so long lines you can split it into new lines, like:

``` js
const dbRefObjectProducts = 
    this.firebase.database()
        .ref('api/v1')
        .child('products')
        .orderByChild('category')
        .equalTo(category);
```

mtrzesniowski commented on app/pages/admin/products/edit/product-edit.service.js:
```diff
@@ -0,0 +1,37 @@
+    updateProduct(inputs, productId) {
+        if(inputs.title && inputs.category && inputs.price && inputs.imageUrl && inputs.amount && inputs.description) {
+            const updateProductData = {
+                'title': inputs.title,
```
You don't need to put key as a string in JS object (if it's not with special chars)

mtrzesniowski commented on app/pages/profile/profile.component.js:
```diff
@@ -0,0 +1,49 @@
+            if (userData) {
+                const user = this.firebase.auth().currentUser;
+
+                if (inputs == null) {
```
why not `if(!inputs)` ?

mtrzesniowski commented on app/pages/profile/profile.service.js:
```diff
@@ -0,0 +1,25 @@
+    }
+
+    updateProfile(inputs, user) {
+        if (inputs.displayName != null && inputs.photoURL) {
```
why not `if (inputs.displayName && inputs.photoURL)`?

Commit:
feature/administration_products_and_categories
change after code review
